### DISCUSSION TITLE
LaTeX export

### DISCUSSION THREAD

**Discussion by rmosvold** [jkitchin/scimax#452] [Q&A]
In my field, we normally use the citation style as defined by the American Psychological Association (APA), but I have not been able to export properly from org-mode to pdf with apastyle in Scimax. In vanilla Emacs, I add the following snippets to my init.el:

`(add-to-list 'org-latex-classes
               '("apa6"
                 "\\documentclass{apa6}"
                 ("\\section{%s}" . "\\section*{%s}")
                 ("\\subsection{%s}" . "\\subsection*{%s}")
                 ("\\subsubsection{%s}" . "\\subsubsection*{%s}")
                 ("\\paragraph{%s}" . "\\paragraph*{%s}")
                 ("\\subparagraph{%s}" . "\\subparagraph*{%s}")))`

I tried to add this to my user.el in Scimax, but the result is still not right, so I assume there must be some export settings that are overriding these for me. Any suggestions about how/where to adjust this?

**Comment by jkitchin** [comment]
Citations and bibliography styles are usually controlled by a bibliography
style that is defined in the org file. It would look something like

\bibliographystyle{apa}

Where apa is the actual style name.



On Tue, May 3, 2022 at 4:09 AM Reidar Mosvold ***@***.***>
wrote:

> In my field, we normally use the citation style as defined by the American
> Psychological Association (APA), but I have not been able to export
> properly from org-mode to pdf with apastyle in Scimax. In vanilla Emacs, I
> add the following snippets to my init.el:
>
> (add-to-list 'org-latex-classes '("apa6" "\\documentclass{apa6}"
> ("\\section{%s}" . "\\section*{%s}") ("\\subsection{%s}" .
> "\\subsection*{%s}") ("\\subsubsection{%s}" . "\\subsubsection*{%s}")
> ("\\paragraph{%s}" . "\\paragraph*{%s}") ("\\subparagraph{%s}" .
> "\\subparagraph*{%s}")))
>
> I tried to add this to my user.el in Scimax, but the result is still not
> right, so I assume there must be some export settings that are overriding
> these for me. Any suggestions about how/where to adjust this?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jkitchin/scimax/discussions/452>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMJCVRGENGQCRSE45RXTNTVIDNLDANCNFSM5U6FXMXA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
John

-----------------------------------
Professor John Kitchin (he/him/his)
Doherty Hall A207F
Department of Chemical Engineering
Carnegie Mellon University
Pittsburgh, PA 15213
412-268-7803
@johnkitchin
http://kitchingroup.cheme.cmu.edu

**Reply by rmosvold** [reply]
I added the following two lines toward the end of my org file:
bibliographystyle:apacite
bibliography:~/references.bib

I also added the following to the preamble of the file:
#+Latex_CLASS: apa6
#+OPTIONS: toc:nil author:nil
#+LaTeX_CLASS_OPTIONS: [a4paper]
#+LaTeX_CLASS_OPTIONS: [doc]
#+LaTeX_HEADER: \author{author name}
#+LaTeX_HEADER: \affiliation{affiliation}
#+LaTeX_HEADER: \leftheader{With the apa6 class, it is good idea to customize the left header here}
#+LaTeX_HEADER: \shorttitle{Your short title goes here}
#+LaTeX_HEADER: \usepackage{breakcites}
#+LaTeX_HEADER: \usepackage{apacite}
#+LaTeX_HEADER: \usepackage{natbib}
#+LaTeX_HEADER: \usepackage{paralist}
#+LaTeX_HEADER: \let\itemize\compactitem
#+LaTeX_HEADER: \let\description\compactdesc
#+LaTeX_HEADER: \let\enumerate\compactenum
#+LaTeX_HEADER: \abstract{Your abstract goes here}
#+LaTeX_HEADER: \keywords{Your keywords go here}

When I export to pdf in vanilla Emacs, I get a file that looks exactly the way it should, but when I export the same file on the same machine, with the exact same settings in init.el (emacs) and user.el (scimax) in scimax, I get a pdf that does not at all look the way it should. That's why I was wondering if there are some general export settings in scimax that are overriding the ones I am trying to use...

**Comment by jkitchin** [comment]
There are probably some default packages that get inserted.

See
http://doc.endlessparentheses.com/Var/org-latex-classes.html

You can turn that off in your apa class with something like
[NO-DEFAULT-PACKAGES]

On Tue, May 3, 2022 at 7:54 AM Reidar Mosvold ***@***.***>
wrote:

> I added the following two lines toward the end of my org file:
> bibliographystyle:apacite
> bibliography:~/references.bib
>
> I also added the following to the preamble of the file:
> #+Latex_CLASS: apa6
> #+OPTIONS: toc:nil author:nil
> #+LaTeX_CLASS_OPTIONS: [a4paper]
> #+LaTeX_CLASS_OPTIONS: [doc]
> #+LaTeX_HEADER: \author{author name}
> #+LaTeX_HEADER: \affiliation{affiliation}
> #+LaTeX_HEADER: \leftheader{With the apa6 class, it is good idea to
> customize the left header here}
> #+LaTeX_HEADER: \shorttitle{Your short title goes here}
> #+LaTeX_HEADER: \usepackage{breakcites}
> #+LaTeX_HEADER: \usepackage{apacite}
> #+LaTeX_HEADER: \usepackage{natbib}
> #+LaTeX_HEADER: \usepackage{paralist}
> #+LaTeX_HEADER: \let\itemize\compactitem
> #+LaTeX_HEADER: \let\description\compactdesc
> #+LaTeX_HEADER: \let\enumerate\compactenum
> #+LaTeX_HEADER: \abstract{Your abstract goes here}
> #+LaTeX_HEADER: \keywords{Your keywords go here}
>
> When I export to pdf in vanilla Emacs, I get a file that looks exactly the
> way it should, but when I export the same file on the same machine, with
> the exact same settings in init.el (emacs) and user.el (scimax) in scimax,
> I get a pdf that does not at all look the way it should. That's why I was
> wondering if there are some general export settings in scimax that are
> overriding the ones I am trying to use...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jkitchin/scimax/discussions/452#discussioncomment-2679716>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMJCVXG6FQUU5C5V64GU3DVIEHYJANCNFSM5U6FXMXA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
John

-----------------------------------
Professor John Kitchin (he/him/his)
Doherty Hall A207F
Department of Chemical Engineering
Carnegie Mellon University
Pittsburgh, PA 15213
412-268-7803
@johnkitchin
http://kitchingroup.cheme.cmu.edu

**Comment by jkitchin** [comment]
I guess what you want is:

```
(add-to-list 'org-latex-classes '("apa6" "\\documentclass{apa6}
[NO-DEFAULT-PACKAGES]
[PACKAGES]" ("\\section{%s}" . "\\section*{%s}") ("\\subsection{%s}" . "\\subsection*{%s}") ("\\subsubsection{%s}" . "\\subsubsection*{%s}") ("\\paragraph{%s}" . "\\paragraph*{%s}") ("\\subparagraph{%s}" . "\\subparagraph*{%s}")))
```

**Reply by rmosvold** [reply]
Sorry that I did not thank you properly for your help and suggestions! When your suggestion also didn't work the way I expected, I realized that I probably should spend some time trying to understand more about how exports work in scimax and the most recent version of org-ref. I have now copied a csl-file with APA 7th version to the /scimax/org-ref/citeproc/csl-styles/ folder, and I then changed the default csl-style to this in org-ref-export.el. By doing this, I got a very close to perfect result with org-ref-export to odt (and pdf) without any other header arguments than these:

#+TITLE: Some title
#+AUTHOR: My Name
#+OPTIONS: toc:nil 
#+LaTeX_CLASS_OPTIONS: [a4paper]
#+LaTeX_CLASS_OPTIONS: [doc]

The only minor issue was that org-ref-export didn't appear to make a distinction between a reference that was given as cite:&something, citep:&something or citeyearpar:&something. Every reference appeared in the text as would cite:&something, or in APA as (Author, year). So, I get closer and closer, hopefully doing something right along the way :-)

**Comment by jkitchin** [comment]
I am glad that was helpful. There will always be some disparity between the bibtex/biblatex types and what CSL supports. I doubt there will ever be feature parity; bib(la)tex is much older and more mature than CSL, and they don't exactly have the same goals. What you are seeing is probably that CSL doesn't have support for citeyearpar.  You can get pretty close on standard types though.
